app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationActionService.scala

```diff
@@ -57,48 +57,46 @@ class ApplicationActionService @Inject() (
   def toApiSubscriptionStatusList(
       application: ApplicationWithSubscriptionData,
       subscriptionFieldDefinitions: Map[ApiContext, Map[ApiVersionNbr, Map[FieldName, SubscriptionFieldDefinition]]],
-      summaryApiDefinitions: Map[ApiContext, ApiData]
+      summaryApiDefinitions: List[ApiDefinition]
     ): List[APISubscriptionStatus] = {
 
-    def handleContext(context: ApiContext, cdata: ApiData): List[APISubscriptionStatus] = {
-      def handleVersion(version: ApiVersionNbr, vdata: ApiVersion): APISubscriptionStatus = {
+    def handleContext(apiDefinition: ApiDefinition): List[APISubscriptionStatus] = {
+      def handleVersion(apiVersionNbr: ApiVersionNbr, apiVersion: ApiVersion): APISubscriptionStatus = {
         def zipDefinitionsAndValues(): List[SubscriptionFieldValue] = {
-          val fieldNameToDefinition = subscriptionFieldDefinitions.getOrElse(context, Map.empty).getOrElse(version, Map.empty)
-          val fieldNameToValue      = application.subscriptionFieldValues.getOrElse(context, Map.empty).getOrElse(version, Map.empty)
+          val fieldNameToDefinition = subscriptionFieldDefinitions.getOrElse(apiDefinition.context, Map.empty).getOrElse(apiVersionNbr, Map.empty)
+          val fieldNameToValue      = application.subscriptionFieldValues.getOrElse(apiDefinition.context, Map.empty).getOrElse(apiVersionNbr, Map.empty)
 
           fieldNameToDefinition.toList.map {
             case (n, d) => SubscriptionFieldValue(d, fieldNameToValue.getOrElse(n, FieldValue.empty))
           }
         }
 
         APISubscriptionStatus(
-          name = cdata.name,
-          serviceName = cdata.serviceName,
-          context = context,
-          apiVersion = vdata,
-          subscribed = application.subscriptions.contains(ApiIdentifier(context, version)),
+          name = apiDefinition.name,
+          serviceName = apiDefinition.serviceName,
+          context = apiDefinition.context,
+          apiVersion = apiVersion,
+          subscribed = application.subscriptions.contains(ApiIdentifier(apiDefinition.context, apiVersionNbr)),
           requiresTrust = false, // Because these are filtered out
           fields = SubscriptionFieldsWrapper(
             applicationId = application.application.id,
             clientId = application.application.clientId,
-            apiContext = context,
-            apiVersion = version,
+            apiContext = apiDefinition.context,
+            apiVersion = apiVersionNbr,
             fields = zipDefinitionsAndValues()
           ),
-          isTestSupport = cdata.isTestSupport
+          isTestSupport = apiDefinition.isTestSupport
         )
       }
 
       val orderDescending: Ordering[ApiVersionNbr] = (x: ApiVersionNbr, y: ApiVersionNbr) => y.value.compareTo(x.value)
 
-      cdata.versions.toList.sortBy(_._1)(orderDescending).map {
-        case (k, v) => handleVersion(k, v)
+      apiDefinition.versions.toList.sortBy(_._1)(orderDescending).map {
```

Comment:
Probably rewrite to drop versionNbr (see comment above) and sort using the version ordering (then .reversed IIRC as it's ascending)